File: WaterVapourInterface/GenerateData.m
% this octave script generate the file data.lammps to be read from lammps input script. it contains a slack of water molecule

clear all
close all

cptmol=0;
cptatom=0;
cptbond=0;
cptangle=0;

txlo=-10; txhi=-txlo;
tylo=-10; tyhi=-tylo;
tzlo=-30; tzhi=-tzlo;

dx=3.4;

PnmpA=load('./WaterMolecule/Position.dat');
BnmpA=load('./WaterMolecule/Bond.dat');
AnmpA=load('./WaterMolecule/Angle.dat');

x=txlo+dx/4;
y=tylo+dx/4;
z=tzlo/2;
while z<=tzhi/2
	while y < tyhi-dx/2
		while x < txhi-dx/2
			cptmol=cptmol+1; 
			for ii=1:length(BnmpA(:,1))
				cptbond=cptbond+1;
				B(cptbond,:)=[cptbond BnmpA(ii,2) BnmpA(ii,3)+cptatom BnmpA(ii,4)+cptatom];
			end
			for ii=1:length(AnmpA(:,1))
				cptangle=cptangle+1;
				Ag(cptangle,:)=[cptangle AnmpA(ii,2) AnmpA(ii,3)+cptatom AnmpA(ii,4)+cptatom AnmpA(ii,5)+cptatom];
			end
			for ii=1:length(PnmpA(:,1))
				cptatom=cptatom+1;
				A(cptatom,:)=[cptatom cptmol PnmpA(ii,3) PnmpA(ii,4) PnmpA(ii,5)+x PnmpA(ii,6)+y PnmpA(ii,7)+z];
			end
			x=x+dx;
		end
		x=txlo+dx/4;
		y=y+dx;
	end
	y=tylo+dx/4;
	z=z+dx;	
end


fid = fopen('data.lammps','wt');
fprintf(fid, '# System\n\n');
fprintf(fid, num2str(cptatom));
fprintf(fid, ' atoms\n');
fprintf(fid, num2str(cptbond));
fprintf(fid, ' bonds\n');
fprintf(fid, num2str(cptangle));
fprintf(fid, ' angles\n');
fprintf(fid, num2str(2));
fprintf(fid, ' atom types\n');
fprintf(fid, num2str(1));
fprintf(fid, ' bond types\n');
fprintf(fid, num2str(1));
fprintf(fid, ' angle types\n');
fprintf(fid, 'extra bond per atom ');
fprintf(fid, num2str(2));
fprintf(fid, '\n');
fprintf(fid, 'extra angle per atom ');
fprintf(fid, num2str(1));
fprintf(fid, '\n');
fprintf(fid, 'extra special per atom ');
fprintf(fid, num2str(2));
fprintf(fid, '\n');
fprintf(fid, num2str([txlo txhi]));
fprintf(fid, ' xlo xhi\n');
fprintf(fid, num2str([tylo tyhi]));
fprintf(fid, ' ylo yhi\n');
fprintf(fid, num2str([tzlo tzhi]));
fprintf(fid, ' zlo zhi\n\n');
fprintf(fid, 'Atoms\n\n');
for ii=1:length(A(:,1))
	for jj=1:7
		fprintf(fid, num2str(A(ii,jj)));
		fprintf(fid, '	');
	end
	fprintf(fid, '\n');
end
fprintf(fid, '\n');
fprintf(fid, 'Bonds\n\n');
for ii=1:length(B(:,1))
	for jj=1:4
		fprintf(fid, num2str(B(ii,jj)));
		fprintf(fid, '	');
	end
	fprintf(fid, '\n');
end
fprintf(fid, '\n');
fprintf(fid, 'Angles\n\n');
for ii=1:length(Ag(:,1))
	for jj=1:5
		fprintf(fid, num2str(Ag(ii,jj)));
		fprintf(fid, '	');
	end
	fprintf(fid, '\n');
end
fclose(fid);

File: solid-NaCl-under-compression/generatedata.py
#!/usr/bin/env python
# coding: utf-8

# # Create a data file for LAMMPS
# ## The system consist of a NaCl crystal wall in contact with a gas of Ethanol molecules
# ## Import libraries
import numpy as np
import copy

# ## parameter choice
dnacl = 2.84 # Na-Cl typical distance
dx, dy, dz = dnacl, dnacl, dnacl
nx, ny, nz = 10, 10, 10
Lx, Ly, Lz = nx*dx, ny*dy, nz*dz
print('The desired Na-Cl layer dimensions are '+str(Lx)+' A x ' +str(Ly)+' A x '+str(Lz)+' A')


# ## Lammps box size
txlo, txhi = -Lx/2, Lx/2
tylo, tyhi = -Ly/2, Ly/2
tzlo, tzhi = -Lz/2, Lz/2

# ## create the NaCl wall
basestructure = np.loadtxt('../../shared/NaCl/Position.dat') # import the 8 basic atoms to replicates
# replicate the initial structure
naclwall = copy.deepcopy(basestructure)
for xx in np.arange(txlo+dx/2,txhi,2*dx):
    for yy in np.arange(tylo+dy/2,tyhi,2*dy):
        for zz in np.arange(tzlo+dz/2,tzhi,2*dz):
            naclwall = np.append(naclwall,basestructure+[0,0,0,0,xx,yy,zz], axis=0)
naclwall = naclwall[8:]
# renumber atoms ids
for n in range(len(naclwall)):
    naclwall[n,0] = np.int64(n+1)

# ## create data lammps file
cptatm = 0
cptmol = 1
atoms = np.zeros((1000000,7))
# ## place the NaCl
for m in naclwall:
    atoms[cptatm] = m[0], cptmol, m[2], m[3], m[4], m[5], m[6]
    cptatm += 1

# ## remove excess lines
atoms = atoms[0:cptatm]       

# ## write LAMMPS data file
f = open("data.lammps", "w")
f.write('# LAMMPS data file \n\n')
f.write(str(cptatm)+' atoms\n')
f.write('\n')
f.write(str(int(2))+' atom types\n')
f.write('\n')
f.write(str(txlo)+' '+str(txhi)+' xlo xhi\n')
f.write(str(tylo)+' '+str(tyhi)+' ylo yhi\n')
f.write(str(tzlo)+' '+str(tzhi)+' zlo zhi\n')
f.write('\n')
f.write('Atoms\n')
f.write('\n')
for nlin in range(len(atoms)):
    newline = atoms[nlin]
    for col in range(len(newline)):
        if col < 3:
            f.write(str(int(newline[col]))+' ')
        else :
            f.write(str(newline[col])+' ')
    f.write('\n')
f.close()

